app/controllers/content_pages_controller.rb

```diff
@@ -12,12 +12,9 @@ def show; end
 
   # GET /content_pages/new
   def new
-    @content_page = ContentPage.new
     # If the new page is a child, pass through its parent id
-    if params[:parent_id]
-      @content_page.parent_id = params[:parent_id]
-    end
-    @content_page
+    # Pages with a nil parent_id are top_level
```

Comment:
I don't think this comment is required - it doesn't make the code clearer and It doesn't reason why the code is required.

Reply:
I left the comment in on purpose.  I think it *emphasises* the use of nil to mean top level.  

Sure, people can figure that out, but they might not notice at first that the parent_id is being used for two purposes, to signify top level pages and to associate children with parents.

But I do think that telling a story is more polite, rather than leaving them to reverse engineer it !

Can you tell that I really don't buy this ruby dogma of no comments in the code !